core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -28,6 +28,64 @@ oppia.constant(
 oppia.constant(
   'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE', '/explorationsummarieshandler/data');
 
+// Sorts the node list for display in collection editor frontend. The list
+// starts with a startNode which contains no prerequisite skills. After that
+// nodes are added to the list if their prerequisite skills match the acquired
+// skills of the last node that was added to the node list.
+oppia.filter('customSorter', function() {
+  return function(nodes) {
+    // Get starting node
+    var startingNode;
+    for (var i = 0; i < nodes.length; i++) {
+      if (nodes[i].getPrerequisiteSkillList().isEmpty()) {
+        startingNode = nodes[i];
+        break;
+      }
+    }
+
+    // Populate node list
+    var nodeList = [];
+    nodeList.push(startingNode);
+    while (true) {
```

Comment:
Yeah that makes sense. At the moment...my algorithm assumes no branching. I could either code for the general case that has branching or throw an error in case branching happens. Any preference on what path to follow?